File: backend/internal/domain/facility/control_cabinet.go
package facility

import (
	"github.com/besart951/go_infra_link/backend/internal/domain"
	"github.com/google/uuid"
)

type ControlCabinet struct {
	domain.Base
	BuildingID       uuid.UUID `gorm:"type:uuid;not null;index;uniqueIndex:idx_cabinet_building_nr"`
	Building         Building  `gorm:"foreignKey:BuildingID"`
	ControlCabinetNr *string   `gorm:"uniqueIndex:idx_cabinet_building_nr"`

	SPSControllers []SPSController `gorm:"foreignKey:ControlCabinetID"`
}

File: frontend/src/lib/domain/facility/control-cabinet.ts
/**
 * Control Cabinet domain types
 * Mirrors backend: internal/domain/facility/control_cabinet.go
 */

import type { Pagination } from '../utils/index.ts';

export interface ControlCabinet {
	id: string;
	control_cabinet_nr: string;
	building_id: string;
	created_at: string;
	updated_at: string;
}

export interface CreateControlCabinetRequest {
	control_cabinet_nr: string;
	building_id: string;
}

export interface UpdateControlCabinetRequest {
	id: string;
	control_cabinet_nr?: string;
	building_id?: string;
}

export interface ControlCabinetListParams {
	page?: number;
	limit?: number;
	search?: string;
	building_id?: string;
}

export interface ControlCabinetListResponse extends Pagination<ControlCabinet> {}

export interface ControlCabinetDeleteImpact {
	control_cabinet_id: string;
	sps_controllers_count: number;
	sps_controller_system_types_count: number;
	field_devices_count: number;
	bacnet_objects_count: number;
	specifications_count: number;
}
